lib/entities.ts

```diff
@@ -100,17 +109,15 @@ function processEvent(store: Store<HassEntities>, updates: StatesUpdates) {
           entityState.last_updated = new Date(toAdd.lu * 1000).toISOString();
         }
         if (toAdd.a) {
-          entityState.attributes = { ...entityState.attributes, ...toAdd.a };
+          Object.assign(attributes, toAdd.a);
         }
       }
-      if (toRemove) {
-        const attributes = { ...entityState.attributes };
-        for (const key in toRemove.a) {
-          delete attributes[key];
-        }
+      if (toRemove?.a) {
+        toRemove.a.forEach((key) => delete attributes[key]);
```

Comment:
I prefer `for …of` over `forEach`.

```suggestion
        for (const key of toRemove.a) {
          delete attributes[key];
        }
```